### PR TITLE
Add sequence number and sorting by sequence or code

### PR DESCRIPTION
**Pull request by marianamarasoiu**
Demo: https://www.cl.cam.ac.uk/~mcm79/codav2_demos/sorting-2018-10-30/?dataset=reach_demo

Still working on tidying it up a little, but do let me know if you have any thoughts :)

Issue: #113 
Later edit: should also fix issue #126

### CODE REVIEW

**Comment by lukechurch** [comment]
We seem to have some test failures - are they expected?

File: data_tools/sample_messages.json [hunk 1]
```diff
@@ -1,30 +1,35 @@
 [
   {
     "MessageID"           : "msg-1",
+    "SequenceID"          : 1,
```
Comment (lukechurch) [inline data_tools/sample_messages.json]: Zero index pls

File: data_tools/validate_message_structure.py [hunk 2]
```diff
@@ -97,7 +107,7 @@ def verify_message(message):
 
 
 def check_string(s):
-    assert isinstance(s, str)
+    assert isinstance(s, unicode)
```
Comment (lukechurch) [inline data_tools/validate_message_structure.py]: I'm not convinced that this is the right change

File: docs/data_formats.md [hunk 3]
```diff
@@ -36,6 +36,7 @@ Message definition (JSON, named $ID)
 ```
 {
   "MessageID"           : String,           // Required, Globally unique
+  "SequenceNumber"      : Int               // Required, Unique, but not necessarily consecutive
```
Comment (lukechurch) [inline docs/data_formats.md]: So if we do this, we're going to have to build a schema upgrade tool - as the data in Firestore production doesn't have this at the moment. I think we should probably go ahead and do that

Comment (marianamarasoiu) [reply docs/data_formats.md]: Yeah, sounds good, I'll open an issue for the schema upgrade tool.

**Comment by marianamarasoiu** [comment]
Thanks for the review! I've fixed the test issues - I needed to update the test data and html file as well.

The "jump to first uncoded message" for vertical coding was broken for a while (or never actually worked), so I've fixed that too.

I've updated the demo - PTAL and let me know if there are any other issues!

**Comment by marianamarasoiu** [comment]
@lukechurch this should be ready for you to have another look at, thanks!

**Comment by lukechurch** [comment]
@marianamarasoiu Sorry to keep this outstanding, it requires a schema upgrade which I'm a bit reluctant to do on top of active projects. I need to think a bit about ensuring a safe way to do this systematically

**Comment by marianamarasoiu** [comment]
@lukechurch This should now be ready to be reviewed again.
I've also updated the demo if you want to try it: https://www.cl.cam.ac.uk/~mcm79/codav2_demos/sorting-2018-10-30/?dataset=reach_demo

Thanks!

**Pull request merged**